docs/decisions/0007-optional-fields.rst

```diff
@@ -0,0 +1,44 @@
+6. Event schema serialization and evolution
+===========================================
+
+Status
+------
+
+Provisional
+
+Context
+-------
+The ``attrs`` documentation specifically defines the notion of optionality as, "An optional attribute is one which
+can be set to None in addition to satisfying the requirements of the sub-validator." Importantly, under this definition,
+the user must still provide a value for this optional field when instantiating the ``attrs``-decorated class,
+but this value can be ``None``. Separately, attrs fields can have a ``default`` value, which will provide a value for
+the field if the user does not supply one at instantiation.
+
+The Avro documentation describes somewhat similar mechanics for its fields. If a field's ``type`` list includes
+"null," the user can set the value of the field to null/None. Separately, if a field is defined with a ``default``
+parameter, then the user does not have to set the value at all. However, it seems the ``default`` parameter may not be
+fully-supported with Kafka (see https://github.com/confluentinc/kafka-rest/issues/427 ). The Avro documentation,
+similar to that of ``attrs``, uses the default parameter to define optionality.
+
+Decision
+--------
+
+- The bridge will convert ``attrs`` fields with a default value of None to Avro fields with type = ["null",<type>]
+and default = "null".
+- Top-level fields in event data definitions will not be optional. This is contrary to and overrides the decision made in
```

Comment:
I need to look more closely at this. I don’t think we are overriding a decision around the top-level fields. I think we simply haven’t implemented any capability around that, which was also true before.

Reply:
there were some capabilities around it before the previous PR that changed the bridge to work with entire signals, but they were removed in favor of getting the basics of the bridge out faster.